### PR TITLE
MCMC code for DM

### PR DESCRIPTION
**Pull request by profxj**
As titled.

### CODE REVIEW

File: frb/figures/dm.py [hunk 1]
```diff
@@ -266,6 +269,7 @@ def fig_cosmic(frbs, clrs=None, outfile=None, multi_model=False, no_curves=False
             Max redshift for the MR line
 
     Returns:
+        dict:
```
Comment (SunilSimha) [inline frb/figures/dm.py:272]: And also ax optionally. Also, dict requires description.

File: frb/dm/mcmc.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+    # Return
+    return PDF_grid
+
+
+def all_prob(Obh70, F, in_DM_FRBp, z_FRB, mu=150., lognorm_s=1.,
+             lognorm_floor=0., beta=3.):
+    """
+
+    Args:
+        Obh70:
```
Comment (SunilSimha) [inline frb/dm/mcmc.py:180]: Args require descriptions,

File: frb/dm/mcmc.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+    # Other info
+    parm_dict['Other'] = dict(DM_MWhalo=50., zscale_host=True, floor=0.)
+    # Return    
+    return parm_dict
+
+def one_prob(Obh70, F, DM_FRBp, z_FRB, mu=150., lognorm_s=1.,
+             lognorm_floor=0., orig=False, beta=4.):
+    """
+
+    Args:
```
Comment (SunilSimha) [inline frb/dm/mcmc.py:86]: Args and return values require descriptions and data types.

File: frb/dm/mcmc.py [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+        all_PDF_Cosmic, axis=0)
+    log_like = np.sum(np.log(Prob))
+
+    # Return
+    return log_like
+
+
+
+@as_op(itypes=[tt.dscalar, tt.dscalar, tt.dscalar, tt.dscalar], otypes=[tt.dvector])
+def calc_likelihood_four_beta3(Obh70, F, mu, lognorm_s):
```
Comment (SunilSimha) [inline frb/dm/mcmc.py:244]: Docstring needs completion,

File: frb/dm/mcmc.py [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+        # The loop above is much faster for only ~10 events
+        ln_like = all_prob(Obh70, F, frb_DMs, frb_zs, #frb.DM.value - frb.DMISM.value, frb.z,
+                    mu=mu, lognorm_s=lognorm_s, lognorm_floor=lognorm_floor, beta=3.)
+
+    # Return
+    return np.array([ln_like])   # Should be log
+
+
+def pm_four_parameter_model(parm_dict:dict, tight_ObH=False, beta=3.):
+    """[summary]
```
Comment (SunilSimha) [inline frb/dm/mcmc.py:278]: Docstring requires completion.

**Review by SunilSimha — CHANGES_REQUESTED** [review]
Minor docstring edits requested.

File: frb/figures/dm.py (hunk 1 above)
Comment (profxj) [reply frb/figures/dm.py:272]: added

File: frb/dm/mcmc.py (hunk 2 above)
Comment (profxj) [reply frb/dm/mcmc.py:180]: Added

File: frb/dm/mcmc.py (hunk 3 above)
Comment (profxj) [reply frb/dm/mcmc.py:86]: added

File: frb/dm/mcmc.py (hunk 4 above)
Comment (profxj) [reply frb/dm/mcmc.py:244]: done

File: frb/dm/mcmc.py (hunk 5 above)
Comment (profxj) [reply frb/dm/mcmc.py:278]: added

**Review by SunilSimha — APPROVED** [review]
Looks good to merge

**Pull request merged**